### PR TITLE
testing_codelab null safe migration + desktop runners

### PR DESCRIPTION
**Pull request by domesticmouse**

### CODE REVIEW

**Comment by domesticmouse** [comment]
Hey @abd99, if you have half an hour to spot check this null safety migration I'd be most appreciative =)

**Comment by abd99** [comment]
Hey @domesticmouse, thanks for pinging me on this!

I thought that we are waiting until all the dependencies get migrated. Are we moving forward with `unsound null-safety` for now because the null-safe version of the `integration_test` package isn't on stable yet?

**Comment by abd99** [comment]
@domesticmouse, I just checked. It isn't working with unsound null-safety either. I'm getting a batch of similar errors:
`Error: A library can't opt out of null safety by default, when using sound null safety.`

I think we'll have to put a hold on this before the required dependencies are in stable.
Should we ask the team if the migrated packages can be released in the next hotfix or something? I believe they are already migrated on master. The next stable release would normally take months.

**Comment by abd99** [comment]
In case you don't know or forgot, the `integration_test` package is now a part of the `flutter/flutter` repo itself.

**Comment by domesticmouse** [comment]
I'm curious where you are seeing the errors, as the CI is all green. What are we not testing in the CI pipeline?

As for holding off this change until all the bits land in stable sounds like the right approach. I believe both of the outstanding packages are in master, but it won't be until next quarter that we see it in stable.

Thoughts @RedBrogdon?

**Comment by abd99** [comment]
> I'm curious where you are seeing the errors, as the CI is all green. What are we not testing in the CI pipeline?

I got those errors on my local environment. We aren't running integration tests on the CI pipeline. I think we should do something about it. We always won't be able to manually test it. I'll try setting up an emulator on GitHub actions and try running integration tests at least in debug mode.

**Comment by domesticmouse** [comment]
Something for a separate PR methinks. Having unchecked tests sounds bad. 😎

**Review by abd99 — CHANGES_REQUESTED** [review]
Thanks a lot for this PR, @domesticmouse! 

A few changes are needed. Before that:

Please change
```
integration_test: ^1.0.1
```

to

```
integration_test: 
    sdk: flutter 
```


You can also remove the `test_driver` directory from `step_07` and the `flutter_driver` dependencies from all the steps.

Let me know if you need any more clarification. This was my first real review =)

**Comment by abd99** [comment]
> Something for a separate PR methinks. Having unchecked tests sounds bad. 😎

I tried setting up an emulator on GitHub Actions, but couldn't get the test to run on it. I think we'll have to discuss about this in our upcoming meeting.

**Comment by domesticmouse** [comment]
> I think we'll have to discuss about this in our upcoming meeting.

Sweet =)

File: testing_codelab/step_03/macos/.gitignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Flutter-related
```
[the remark this answers is not in the document]
Comment (domesticmouse) [reply testing_codelab/step_03/macos/.gitignore:1]: We have a mission to make codelabs available on all platforms that they work. So yeah, we kinda do. =)

**Comment by domesticmouse** [comment]
Hey @abd99, is this now safe to land?

**Comment by abd99** [comment]
LGTM, thanks!

**Review by abd99 — APPROVED** [review]

**Pull request merged**